### DISCUSSION TITLE
How to migrate all data from the default H2 database to a PostgreSQL database?

### DISCUSSION THREAD

**Discussion by mirawara** [keycloak/keycloak#26871] [Q&A]
In my previous installations of Keycloak, I didn't use an external database. However, now I want to change this, but I can't find a satisfactory guide. Is there an elegant and secure way to transfer all the data, including the user federations? Currently, I'm using Keycloak container v.20.0.0.

Thank you in advance.

**Comment by stianst** [comment]
Take a look at the import/export stuff. That can export to json, and you can then reimport to postfres

**Reply by mirawara** [reply]
Thank you for your response.
That's what I've tried, but not all realms are imported correctly, for example those with users imported from an LDAP server. That's why I would have preferred a complete migration from H2. Is it possible that there is no way to dump the database?
